shopinvader_payment/components/payment_transaction_event_listerner.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo.addons.base_rest.controllers.main import _PseudoCollection
+from odoo.addons.component.core import Component
+from odoo.addons.shopinvader import shopinvader_response
+
+
+class SaleOrderPaymentTransactionEventListener(Component):
+    _name = "sale.order.payment.transaction.event.listener"
+    _inherit = "base.event.listener"
+    _apply_on = ["sale.order"]
+
+    def _confirm_and_invalidate_session(self, sale_order):
+        shopinvader_backend = sale_order.shopinvader_backend_id
+        if not shopinvader_backend:
+            return
+        sale_order.action_confirm_cart()
+        response = shopinvader_response.get()
+        response.set_session("cart_id", 0)
+        response.set_store_cache("cart", {})
+        # TODO we should not have to return the last_sale information into the
```

Comment:
I agree we should only return the id of the sale order.
And we even not need to keep in cache this information because we only show once the last_sale.
I will talk with @thibaultrey to see what he need on front side